src/js/stores/MesosStateStore.js

```diff
@@ -287,19 +288,23 @@ class MesosStateStore extends GetSetBaseStore {
     // Combine framework (if matching framework was found) and filtered
     // Marathon tasks. This will give you a list of framework tasks including
     // the scheduler tasks or a list of Marathon application tasks.
-    return frameworks.reduce(function(serviceTasks, framework) {
-      const { tasks = [], completed_tasks = {}, name } = framework;
+    return frameworks.reduce(function(serviceTasks, frameworkProperties) {
+      const framework = new Framework(frameworkProperties);
       // Include tasks from framework match, if service is a Framework
-      if (service instanceof Framework && name === serviceName) {
+      if (
+        service instanceof Framework &&
+        lrtrim(framework.name, "/") === lrtrim(service.getId(), "/")
+      ) {
         return serviceTasks
-          .concat(tasks, completed_tasks)
+          .concat(framework.getTasks(), framework.getCompletedTasks())
```

Comment:
@MatApple As a example, what is the implications when accessing direct `framework ["tasks"]` instead of using a getter here?